__tests__/unit/lists/LList.es6

```diff
@@ -0,0 +1,214 @@
+jest.autoMockOff();
+var IM = require('immutable'),
+    SLinkedList = require('../../../src/lists/LList');
+
+describe('SLinkedList', () => {
+  describe('new instance initialization', () => {
+    let sLLEmpty = new SLinkedList();
+
+    it('starts with a null head', () => {
+      expect(sLLEmpty.head).toBeNull();
+    });
+
+    it('starts with a null tail', () => {
+      expect(sLLEmpty.tail).toBeNull();
+    });
+
+    it('starts with an empty _lzStore', () => {
+      expect(sLLEmpty.size).toBe(0);
+    });
+
+    it('stores a number on initialization', () => {
+      let sLLNumber = new SLinkedList(1);
+      expect(sLLNumber.size).toBe(1);
+      expect(sLLNumber.head.data).toEqual(1);
+      expect(sLLNumber.head.next).toBeNull();
+      expect(sLLNumber.tail.data).toEqual(1);
+    });
+
+    it('stores a string on initialization', () => {
+      let sLLString = new SLinkedList('string value');
+      expect(sLLString.size).toBe(1);
+      expect(sLLString.head.next).toBeNull();
+      expect(sLLString.head.data).toEqual('string value');
+      expect(sLLString.tail.data).toEqual('string value');
+      expect(sLLString.tail.next).toBeNull();
+
+    });
+
+    it('stores an object on initialization', () => {
+      let sLLObject = new SLinkedList({ name: 'clark' });
+      expect(sLLObject.size).toBe(1);
+      expect(sLLObject.head.next).toBeNull();
+      expect(sLLObject.tail.next).toBeNull();
+      expect(sLLObject.head.data).toEqual({ name: 'clark' });
+      expect(sLLObject.tail.data).toEqual({ name: 'clark' });
+    });
+
+    it('stores an array on initialization', () => {
+      let sLLArray = new SLinkedList([
+          'random string', 
+          {'asdfasdf':'asdf'}, 
+          { name: 'anna' }, 
+          26, 
+          ['tail value'],
+        ]);
+      expect(sLLArray.size).toBe(5);
+      expect(sLLArray.head.data).toEqual('random string');
+      expect(sLLArray.head.next).toNotEqual(null);
+      expect(sLLArray.head.next.data['asdfasdf']).toBe('asdf');
+      expect(sLLArray.head.next.next.next.data).toBe(26);
+      expect(sLLArray.tail.next).toBeNull();
+      expect(sLLArray.tail.data).toEqual(['tail value']);
+    });
+
+    it('contains immutable nodes', () => {
+      let sLLNumber = new SLinkedList(1);
+      let changeSomething = () =>{
+        sLLNumber.head.data = 2;
+      }
+      expect(changeSomething).toThrow();
+      expect(sLLNumber.head.data).toEqual(1); 
+    });
+  });
+
+  describe('public interface instance methods', () => {
+    let sLLNumber = new SLinkedList(1);
+    let sLLArray = new SLinkedList([
+        'random string', 
+        {'asdfasdf':'asdf'}, 
+        { name: 'anna' },
+      ]);
+
+
+    describe('prepends', () => {
+
+      it('returns a new list when prepending undefined', () => {
```

Comment:
OMGz DAN THIS IS WHAT I SAID IN OUR LAST CHECKIN WHY YOU NO LISTEN

:baby_chick: